### ISSUE TITLE
Client Verbosity Settings

### ISSUE THREAD

**Issue by denelon** [microsoft/winget-cli#161]
# Description of the new feature/enhancement

As a power user I would like to be able to get additional data about what the client is doing so I can be informed.

# Proposed technical implementation details (optional)
Note:
The verbosity levels might be "silent", "default", and "verbose".
There should also be a setting or mechanism to support screen readers
     Progress bars don't work well for screen reader applications

**Comment by josephsl** [comment]
Hi,

Regarding screen readers and progress bars: if we're talking about progress messages when an app is being downloaded/installed/updated/uninstalled, providing a textual output would work, something to the effect of:

winget install notepad++

Downloading 10%
Downloading 20%
...
Downloading 100%
Installing 0%
...
Installing 100%

For a practical example as to how screen readers can react to console output changes, take a look at how Git command-line for Windows does it when fetching commits while Narrator or NVDA is running.

Thanks.

**Comment by vedantmgoyal2009** [comment]
@denelon I guess this is completed. https://github.com/microsoft/winget-cli/blob/master/schemas/JSON/settings/settings.schema.0.2.json#L34

**Comment by Trenly** [comment]
> @denelon I guess this is completed. [`master`/schemas/JSON/settings/settings.schema.0.2.json#L34](https://github.com/microsoft/winget-cli/blob/master/schemas/JSON/settings/settings.schema.0.2.json#L34)

I disagree. This is about verbosity during the install process, not post-install looking at a log.

**Comment by denelon** [comment]
This may also be part of the mechanism to support different UX (User Experience) modes. The default is the way the client operates today where users must type the correct command to get the desired behavior. The other is more of an "interactive" mode where the user sees more prompts and WinGet tries to help them be more successful.

As I've been thinking a bit more about it maybe it's a "prompt" mode where the user sees more prompts or hints when the client can make intelligent suggestions or provide options. This shouldn't be confused with:

```text
Are you sure [Yes | No] ?_
```

It's more like a command was run like "winget upgrade <package>" and the installer type changes. Rather than telling the user they need to uninstall the previous version "manually", the client could prompt:
```text
Would you like to remove the previous version and continue [Yes | No]?_
```

We will clearly have some discoverability concerns with this setting, so maybe there should be a first run experience (if no explicit setting is in place) where the first time the user runs "winget" we can provide an informational message about this mode so it's not breaking for automated scenarios.
